### PR TITLE
Feature/otherSpectralClusterings+MiniBatchKMeans

### PR DESCRIPTION
**Pull request by AHReccese**
#### What does this implement/fix? Explain your changes.
* Add support for SpectralBiclustering
* Add support for SpectralCoclustering
* Add support for MiniBatchKMeans 
* Refactor SpectralClustering test folder

### CODE REVIEW

File: SUPPORTED_MODELS.md [hunk 1]
```diff
@@ -302,6 +302,16 @@
 		<td><b>Spectral Clustering</b></td>
         <td>>=0.4</td>
 	</tr>
+	<tr align="center">
+		<td>5</td>
```
Comment (sadrasabouri) [inline SUPPORTED_MODELS.md:311]: 6

File: SUPPORTED_MODELS.md [hunk 2]
```diff
@@ -302,6 +302,16 @@
...
+		<td><b>Spectral Biclustering</b></td>
+        <td>>=0.4</td>
+	</tr>
+	<tr align="center">
+		<td>5</td>
```
Comment (sadrasabouri) [inline SUPPORTED_MODELS.md:316]: 7

File: SUPPORTED_MODELS.md [hunk 3]
```diff
@@ -287,6 +287,11 @@
 		<td><b>Bisecting Kmeans</b></td>
         <td>>=0.4</td>
 	</tr>
+	<tr align="center">
+		<td>2</td>
```
Comment (sadrasabouri) [inline SUPPORTED_MODELS.md:291]: I think you should update IDs.

**Review by sadrasabouri — CHANGES_REQUESTED** [review]

File: SUPPORTED_MODELS.md (hunk 3 above)
Comment (AHReccese) [reply SUPPORTED_MODELS.md]: my bad, applied.

**Review by sadrasabouri — APPROVED** [review]

**Pull request merged**